### ISSUE TITLE
Failing to create conda env from environment.yml

### ISSUE THREAD

**Issue by CetINV** [klarrieu/RiverREM#12]
I'm not sure if someone of my skill set, or lack thereof should be posting here. 
So if this is out of line please do close or remove. 

No matter which route I take to install RiverREM, I run into a GDAL error that I believe is caused by Python. I cannot get Python 3.6 to install in the RiverREM env. All ways I've tried to install have resulted in Python 3.10 being installed and GDAL either not being found or quitting as posted below. 

The process quits with no error at: 

(riverrem) C:\RiverREM>python
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:30:19) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from riverrem.REMMaker import REMMaker
>>> rem_maker = REMMaker(dem='/rem/data/kafuel.tif', out_dir='/rem/data')
[INFO] 2022-08-25 10:41:23,512 - Getting DEM projection.
[INFO] 2022-08-25 10:41:23,522 - Reading DEM as array.
[INFO] 2022-08-25 10:41:23,618 - Getting DEM bounds.
[INFO] 2022-08-25 10:41:23,618 - Mapping array indices to coordinates.
>>>

Any advice would be very appreciated.

**Comment by klarrieu** [comment]
No worries, sorry you were having a hard time! The good news is that it looks like you did eventually get it installed properly. Your example code snippet doesn't look like it quit without an error message, it looks like it just finished initializing the `rem_maker` object. From there, you can run `rem_maker.make_rem()` and `rem_maker.make_rem_viz(cmap='Blues', z=10)` to create an REM and REM visualization as outlined in the [readme](https://github.com/OpenTopography/RiverREM#usage)/[docs](https://opentopography.github.io/RiverREM/remindex.html) (I've been really liking the combo of the Blues colormap and z=10 vertical exaggeration lately).

That being said, I am curious what the GDAL error you were encountering was. Were you trying to install with Python 3.6 in a new environment or an existing Python 3.6 environment? As mentioned in another issue (https://github.com/OpenTopography/RiverREM/issues/2#issuecomment-1225656479), it can be tricky to get Windows, Anaconda, and GDAL to play nicely sometimes.

**Comment by CetINV** [comment]
Oh my... 🤦‍♂️
My lack of skill was exemplified by opening this non-issue.. sorry 

The GDAL not found error I was receiving was when setting up RiverREM without using the environment.yml. It is the same issue as you linked to. 

Thank you for clarifying that it is working as intended. 
So excited to give this a try with some of the many rivers in my country. 

How well would you say RiverREM works for SRTM data? 

Thank you for being so kind!

**Comment by klarrieu** [comment]
No worries, trust me I've been there too 😅

SRTM works if nothing better is available, but it won't be able to resolve any features smaller than about 30 meters, which is the approximate pixel size for SRTM DEMs. Also note that SRTM comes in a geographic coordinate system (lat/long units) so you might want to first project the DEM raster into a local/projected coordinate system (e.g. UTM, units of meters). 

I actually use a small SRTM dataset to run test cases when building the package, so this will give you an idea of the resolution:

![image](https://user-images.githubusercontent.com/38138126/186721368-71db1a2f-446d-4193-ac7c-162f5d585d54.png)

That is a section of the Crooked River in Smith Rock, Oregon, USA: https://www.google.com/maps/@44.3680628,-121.1295543,14.99z/data=!5m1!1e4

It might look a bit better for larger rivers with widths significantly greater than 30 meters (like Kafue and Zambezi). Hope that helps!

**Comment by CetINV** [comment]
What makes for a good REM with all those undulation in your REMs? Is it purely the terrain? I'm assuming the accuracy of the data is a big factor too. 

I guess I will have to capture some LiDAR datasets of the rivers here. 

What do you think of this? 😁😁😁 I'll still be ecstatic no matter what you think haha. Though I will always appreciate any tips!

<img src="https://i.imgur.com/q9nWnvk.png" width="1012" height="841">
Image: Luangwa River near the Bulonga Mountains, Zambia.

**Comment by klarrieu** [comment]
I think REMs look best for [meandering rivers](https://en.wikipedia.org/wiki/Meander) with relatively flat floodplain areas and low slopes. That's where you're most likely to get river features like [oxbows and scroll bars](https://d3i71xaburhd42.cloudfront.net/151a5b6bad406daa6000f73f578de454558ecf7a/32-Figure2.6-1.png). Resolution is also important and I use 1m resolution lidar data for my example REMs.

Glad you got it working, I think your REM looks great! You might want to play around with adjusting the [colormap (cmap) and vertical exaggeration (z) parameters for the make_rem_viz method](https://opentopography.github.io/RiverREM/remindex.html#riverrem.REMMaker.REMMaker.make_rem_viz). Increasing to z=10 or higher would give it a little more contrast, but overall it looks awesome!

**Comment by CetINV** [comment]
Ah alright. I will try find a couple that will work better. I wish we had a national LiDAR database. 

Thank you!! I think so too 😁 and thank you again for creating RiverREM! I will definitely play with the parameters for visualisation. Now that its working, everything is making sense. 

<img src="https://i.imgur.com/KsbGJCK.jpg">

**Comment by klarrieu** [comment]
Hello, I thought you might be interested in seeing this. I made a REM of the Luangwa River between Kakumbi and Chitungu. I made it with the [Copernicus 30m data](https://portal.opentopography.org/raster?jobId=rt1661873847548) instead of SRTM because it is a DSM, so it removes the vegetation and shows bare earth unlike SRTM. I used the "terrain" colormap and z=10.

![luangwa_cop30_hillshade-color_crop](https://user-images.githubusercontent.com/38138126/187489449-f076b2d5-f52b-4a63-a590-470831eeb77b.png)
